src/test/java/com/artipie/RepoPermissionsFromSettingsTest.java

```diff
@@ -103,6 +110,88 @@ void returnsEmptyMapWhenPermissionsAreNotSet() {
         );
     }
 
+    @Test
+    void updatesUserPermissions() throws IOException {
+        final String repo = "rpm";
+        this.addSettings(
+            repo,
+            new MapOf<String, List<String>>(
+                new MapEntry<>("david", new ListOf<String>("add", "update"))
+            )
+        );
+        final String olga = "olga";
+        final String victor = "victor";
+        final String download = "download";
+        final String deploy = "deploy";
+        new RepoPermissions.FromSettings(new Settings.Fake(this.storage))
+            .addUpdate(
+                repo,
+                new ListOf<>(
+                    new RepoPermissions.UserPermission(olga, new ListOf<>(download, deploy)),
+                    new RepoPermissions.UserPermission(victor, new ListOf<>(deploy))
+                )
+            ).toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            "Added permissions for olga",
+            this.permissionsForUser(repo, olga),
+            Matchers.contains(download, deploy)
+        );
+        MatcherAssert.assertThat(
+            "Added permissions for victor",
+            this.permissionsForUser(repo, victor),
+            Matchers.contains(deploy)
+        );
+    }
+
+    @Test
+    void addsUserPermissionWhenOriginalPermissionsAreNotSet() throws IOException {
+        final String repo = "go";
+        this.addEmpty(repo);
+        final String ann = "ann";
+        final String download = "download";
+        new RepoPermissions.FromSettings(new Settings.Fake(this.storage))
+            .addUpdate(
+                repo,
+                new ListOf<>(new RepoPermissions.UserPermission(ann, new ListOf<>(download)))
+            )
+            .toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            this.permissionsForUser(repo, ann),
+            Matchers.contains(download)
+        );
+    }
+
+    @Test
+    void deletesPermissionSection() throws IOException {
```

Comment:
@olenagerasimova this test will pass if `remove` method just clears all repositories. May we check that other repos stay intact after removing a repo?